client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -63,22 +87,31 @@ tie.factory('PythonPrereqCheckService', [
      *    WRONG_LANGUAGE_ERRORS.
      */
     var detectAndGetWrongLanguageType = function(code) {
-      var codeLines = getNonStringLines(code);
+      var obscuredCode = getObscuredCode(code);
+      var rawCodeArray = code.split('\n');
 
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
-        if (error.allowMultiline) {
-          if (code.search(regexp) !== -1) {
-            return error.errorName;
-          }
-        } else {
-          for (var j = 0; j < codeLines.length; j++) {
-            if (codeLines[j].search(regexp) !== -1) {
-              return error.errorName;
-            }
+        // Lookup character location of error
+        var firstErrorColumnNumber = obscuredCode.search(regexp);
+        if (firstErrorColumnNumber !== -1) {
+          var firstErrorLineNumber = 0;
+
+          // After regex matches an error, loop through codelines
+          for (var l = 0; firstErrorColumnNumber >= 0; l++) {
+            // Subtract l.length from firstErrorCharIndex.
```

Comment:
l is a number, so there's no l.length. I think you mean "Subtract the length of the corresponding line from firstErrorCharIndex" -- except that there's no firstErrorCharIndex either so you'll want to rename firstErrorColumnNumber to firstErrorCharIndex here.

Reply:
Done.